barretenberg/cpp/src/barretenberg/dsl/acir_format/avm_recursion_constraint.cpp

```diff
@@ -168,13 +170,28 @@ AggregationObjectIndices create_avm_recursion_constraints(Builder& builder,
     // proof with public inputs reinserted std::vector<uint32_t> proof_indices =
     //     ProofSurgeon::create_indices_for_reconstructed_proof(input.proof, input.public_inputs);
 
-    std::vector<field_ct> proof_fields;
-    proof_fields.reserve(input.proof.size());
-
-    for (const auto& idx : input.proof) {
-        auto field = field_ct::from_witness_index(&builder, idx);
-        proof_fields.emplace_back(field);
-    }
+    auto fields_from_witnesses = [&](std::vector<uint32_t> const& input) {
+        std::vector<field_ct> result;
+        result.reserve(input.size());
+        for (const auto& idx : input) {
+            auto field = field_ct::from_witness_index(&builder, idx);
+            result.emplace_back(field);
+        }
+        return result;
+    };
+
+    auto const proof_fields = fields_from_witnesses(input.proof);
```

Comment:
Yeah as Facundo said elsewhere, our convention is `const [type] [name]` but it's not a big deal.